pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -423,11 +427,25 @@ func (c *Controller) syncProvisionRequestsPodTemplates(ctx context.Context, requ
 			return err
 		}
 
+		var shouldUpdate bool
+
 		if err == nil {
-			if err := ctrl.SetControllerReference(request, pt, c.client.Scheme()); err != nil {
-				return err
+			if metav1.IsControlledBy(pt, wl) {
```

Comment:
Add a comment
```
// transfer the ownership of the template to the ProvisioningRequest
```
Actually, it seems we could keep the templates as owned by the workload, but since prior to the PR they were owned by ProvisioningRequests I think it makes sense to keep that behavior.

Reply:
Done